### PR TITLE
Fix eclectic docs

### PR DESCRIPTION
**Pull request by sholderbach**
* Make `cargo doc` happy again
* Let the `lib.rs` doc test a few constructor examples.
* Fix badges and repo links in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # A readline replacement written in Rust
 
-![](https://img.shields.io/github/license/jntrnr/reedline) ![](https://img.shields.io/crates/v/reedline.svg) ![](https://docs.rs/reedline/badge.svg?version=0.1.0) ![](https://img.shields.io/github/repo-size/jntrnr/reedline) ![](https://img.shields.io/tokei/lines/github/jntrnr/reedline) ![](https://img.shields.io/github/commit-activity/m/jntrnr/reedline) ![](https://img.shields.io/github/issues-pr-closed-raw/jntrnr/reedline) ![](https://img.shields.io/discord/601130461678272522) ![](https://img.shields.io/twitch/status/jntrnr?style=social)
+![GitHub](https://img.shields.io/github/license/jntrnr/reedline) ![Crates.io](https://img.shields.io/crates/v/reedline) ![docs.rs](https://img.shields.io/docsrs/reedline) ![Discord](https://img.shields.io/discord/601130461678272522) ![Twitch Status](https://img.shields.io/twitch/status/jntrnr?style=social)
```
Comment (mzanrosso) [inline README.md]: Why did you remove some badges ? they were to much ?

**Comment by jntrnr** [comment]
Looks good!
